JASP-Engine/JASP/R/ancova.R

```diff
@@ -1224,491 +712,528 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
+                       label = "Bootstrapping Post Hoc Test")
       
-      levelA <- "."
-      levelB <- "."
-      row <- list("(I)"=levelA, "(J)"=levelB)
-      pTukey <- "."
-      pScheffe <- "."
-      pBonf <- "."
-      pHolm <- "."
-      pSidak <- "."
-      effectSize <- "."
-      md <- "."
-      se  <- "."
-      t <- "."
-      p  <- 1
-      uprBound <- "."
-      lwrBound <- "."
+      ## Computation
+      bootstrapPostHoc <- try(boot::boot(data = dataset, statistic = .bootstrapPostHoc, 
+                                         R = options[["postHocTestsBootstrappingReplicates"]],
+                                         options = options, 
+                                         nComparisons = nrow(resultPostHoc),
+                                         postHocVariablesListV = postHocVariablesListV,
+                                         postHocVarIndex = postHocVarIndex),
+                              silent = TRUE)
       
-      if (length(mainPostResults) == 1){
-        
-      } else if(any(class(mainPostResults) == "try-error")) {
-        
-        .addFootnote(footnotes, 
-                     symbol = "<i>Note.</i>", 
-                     text = "Some comparisons could not be performed. Possibly too few samples.")
-        
-      } else {
-        
-        thisContrast <- unlist(strsplit(as.character(mainPostResults$contrast[i]), split = " - "))
-        levelA <- thisContrast[1]
-        levelB <- thisContrast[2]
-        
-        md <- .clean(as.numeric(mainPostResults$estimate[i]))
-        se <- .clean(as.numeric(mainPostResults$SE[i]))
-        t <- .clean(as.numeric(mainPostResults$t.ratio[i]))
-        
-        lwrBound <- .clean(as.numeric(mainPostResults$lower.CL[i]))
-        uprBound <- .clean(as.numeric(mainPostResults$upper.CL[i]))
-        
-        
-        if (options$postHocTestEffectSize & nrow(dataset) > 0 & length(postHocVariables[[postHocVarIndex]]) == 1) {
-          x <- dataset[(dataset[.v(thisVarName)] == levelA), .v(options$dependent)]
-          y <- dataset[(dataset[.v(thisVarName)] == levelB), .v(options$dependent)]
-          n1 <- length(x)
-          n2 <- length(y)
-          den <- sqrt(((n1 - 1) * var(x) + (n2 - 1) * var(y)) / (n1 + n2 - 2))
-          effectSize <- .clean(md / den)
+      bootstrapSummary <- summary(bootstrapPostHoc)
+
+      ci.fails <- FALSE
+      bootstrapPostHocConf <- t(sapply(1:nrow(bootstrapSummary), function(comparison){
+        res <- try(boot::boot.ci(boot.out = bootstrapPostHoc, conf = options$confidenceIntervalIntervalPostHoc, type = "bca",
+                                 index = comparison)[['bca']][1,4:5])
+        if(!inherits(res, "try-error")){
+          return(res)
+        } else if(identical(attr(res, "condition")$message, "estimated adjustment 'a' is NA")){
+          ci.fails <<- TRUE
+          return(c(NA, NA))
+        } else{
+          return(res)
```

Comment:
I think this should be left as is - if it fails then it will print a footnote, which I think will suffice in this case. If something really wonky goes on in bootstrapping it will be caught by the earlier try-catch/setError construction.

Reply:
The footnote is only added when the error message contains `"estimated adjustment 'a' is NA"`, but the error message might be something completely different. The `else` captures that situation and that situation will sort of crash and burn ;)